pkg/resource/db_cluster_parameter_group/hooks.go

```diff
@@ -173,3 +201,235 @@ func sdkTagsFromResourceTags(
 	}
 	return tags
 }
+
+func equalStrings(a, b *string) bool {
+	if a == nil {
+		return b == nil || *b == ""
+	}
+	return (*a == "" && b == nil) || *a == *b
+}
```

Comment:
Noted.  As you've pointed out in that comment, the issue applies to many different instances of this function across the codebase.  Would you prefer that I fix this one instance, knowing that it will be out of alignment with all of the other, broken instances?

Reply:
Ideally we should fix it now. Let's not wait until it lands in runtime. Updating all the controllers will take alot of time.

Reply:
Turns out that this function was unused, so I just pruned it.